### PR TITLE
Risk change dialog pops up after being dismissed (EXPOSUREAPP-3536)

### PR DESCRIPTION
**Pull request by SamuraiKek**
### Description

This PR fixes a bug where the LiveData from the VM wasn't getting updated anymore, causing a dialog to show up every time you traveled to another screen and then returned.
Issue was caused by LiveData being filtered and once it had a value, the filter was ignoring any other state the LiveData could be in.

How to quickly test:

In RiskLevelTransaction, comment out the start() function and replace it with this one:

```kotlin
suspend fun start() = lockAndExecute(
        scope = transactionScope,
        timeout = RISK_LEVEL_TRANSACTION_TIMEOUT
    ) {
        lastCalculatedRiskLevelScoreForRollback.set(RiskLevelRepository.getLastCalculatedScore())
        executeUpdateRiskLevelScore(INCREASED_RISK)
        executeRiskLevelCalculationDateUpdate()
        executeClose()
        return@lockAndExecute
    }
```

- Build & run the app with the "hacked" version of the start function to put the app in a high risk state.
- Change the INCREASED_RISK to LOW_LEVEL_RISK in executeUpdateRiskLevelScore() and build & run again.
- App should display the dialog with the low risk card in the background.
- Pressing Ok on the dialog dismisses the dialog.
- Navigating to a different screen and returning to the home screen won't result in a new dialog being displayed.

### CODE REVIEW

**Review by AlexanderAlferov — DISMISSED** [review]
Tested with steps in PR description.
Smoke tested.
lgtm

**Review by BMItr — DISMISSED** [review]
Code looks clean to me and good to read :ok_hand:

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,136 +0,0 @@
-<component name="ProjectCodeStyleConfiguration">
```
Comment (harambasicluka) [inline .idea/codeStyles/Project.xml:1]: I think this shouldn't be part of your commit.

**Review by harambasicluka — CHANGES_REQUESTED** [review]
lgtm, but I think the Project.xml shouldn't be part this PR.

**Review by harambasicluka — APPROVED** [review]

**Review by AlexanderAlferov — APPROVED** [review]

**Pull request merged**